### PR TITLE
Fix for strange interaction between features and versions

### PR DESCRIPTION
**Pull request by mcg1969**
Jeff Reback reported some strange version selections in his Pandas CI process. For some reason it was tied to a subtle bug in how I was counting the number of packages with features being installed. I've adjusted the feature package count code a bit and that seems to have done the trick. I must confess I'm not _totally_ clear on why the original code was buggy. But what I removed was simply intended to reduce the complexity of the objective function---a performance hack. What is now here is far simpler and more straightforward.

### CODE REVIEW

File: conda/resolve.py [hunk 1]
```diff
@@ -671,15 +671,13 @@ def generate_feature_count(self, C, trackers):
 
     def generate_feature_metric(self, C, groups, specs):
         eq = {}
+        total = 0
         for name, group in iteritems(groups):
             nf = [len(self.features(fn)) for fn in group]
             maxf = max(nf)
```
Comment (kalefranz) [inline conda/resolve.py]: If we're worried about performance here, why not turn `nf` into a generator.

    maxf = max((len(self.feature(fn) for fn in group)))

Comment (mcg1969) [inline conda/resolve.py]: Oh I'm not worried about the performance of the function itself, rather the performance of the SAT solver that uses its output.

Comment (kalefranz) [inline conda/resolve.py]: Oh, never mind. Didn't see enough of the code.

Comment (mcg1969) [inline conda/resolve.py]: Plus, I use all of those feature lengths in a later step.

**Pull request merged**